### PR TITLE
Filter to scan only Backupjobs

### PR DESCRIPTION
**Pull request by Gab321**
The script now checks only the Backupjobs you can change this behavior with the parameter -type Backup/Backupsync

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Monitor the backup status of a single machine made with veeam
       -vm       NAME of the machince to check (Not the job name)
       -warning  WARNING offset (s) of the successful backup end time [Default: 86400]
       -critical CRITICAL offset (s) of the successful backup end time [Default: 172800]
+	  -type     TYPE of Backup. Choose between Backup or Copy (Jobs) [Default: Backup]
```
Comment (raffis) [inline README.md]: use whitespaces, no tabs.

File: check_vm_backup.ps1 [hunk 2]
```diff
@@ -8,7 +8,8 @@
 param (
    [string]$vm=$FALSE,
    [string]$warning=86400,
-   [string]$critical=172800
+   [string]$critical=172800,
+   [string]$type="Backup"
```
Comment (raffis) [inline check_vm_backup.ps1]: I would prefer everything in lowercase (backup and copy)

File: check_vm_backup.ps1 [hunk 3]
```diff
@@ -18,17 +19,32 @@ if($vm -eq $FALSE) {
...
+    exit 3
+}
 
-foreach($job in (Get-VBRJob)) {
+if($type -eq "Copy") {
+    $type="BackupSync"
+}
+
+if($type -ne "Backup" -and $type -ne "Backupsync") {
+    echo "parameter -type wrong, please check the spelling"
```
Comment (raffis) [inline check_vm_backup.ps1]: echo "parameter -type accepts backup or copy as value"

File: check_vm_backup.ps1 [hunk 4]
```diff
@@ -18,17 +19,32 @@ if($vm -eq $FALSE) {
     exit 3
 }
 
-$now=[Math]::Floor([decimal](Get-Date(Get-Date).ToUniversalTime()-uformat "%s"))
+if($vm -eq $FALSE) {
+    echo "param -type is missing and must either be Backup or Copy. Leave it empty for Backupjobs"
```
Comment (raffis) [inline check_vm_backup.ps1]: Why did you replace " echo "param -vm missing" with this?

File: check_vm_backup.ps1 [hunk 5]
```diff
@@ -18,17 +19,32 @@ if($vm -eq $FALSE) {
...
 
     if(!$session) {
         continue;
     }
- 
-    $object = $Session.GetTaskSessions() | Where-Object {$_.Name -eq $vm}
-    
+
+    $object = $session.GetTaskSessions() | Where-Object {$_.Name -eq $vm}
+
```
Comment (raffis) [inline check_vm_backup.ps1:41]: keep one new line, not two

**Review by raffis — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Gab321) [reply README.md]: Notepad++ showed it right for me. I changed it with a different editor

File: check_vm_backup.ps1 (hunk 2 above)
Comment (Gab321) [reply check_vm_backup.ps1]: changed in the newest commit

File: check_vm_backup.ps1 (hunk 4 above)
Comment (Gab321) [reply check_vm_backup.ps1]: Variable $vm not changed to $type after copying it. I corrected this failure. param -vm missing is still available.

File: check_vm_backup.ps1 (hunk 3 above)
Comment (Gab321) [reply check_vm_backup.ps1]: changed to the comment suggested

File: check_vm_backup.ps1 (hunk 5 above)
Comment (Gab321) [reply check_vm_backup.ps1:41]: added

**Comment by raffis** [comment]
@Gab321 thanks for pr.

**Pull request merged**